internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -126,16 +126,17 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	// ignore unsupported platforms
-	if !isSupportedPlatform(clusterDeployment) {
+	
+	if clusterDeployment.Spec.ClusterInstallRef == nil {
+		log.Infof("AgentClusterInstall not set for ClusterDeployment %s", clusterDeployment.Name)
 		return ctrl.Result{}, nil
 	}
 
-	clusterInstall := &hiveext.AgentClusterInstall{}
-	if clusterDeployment.Spec.ClusterInstallRef == nil {
-		log.Infof("AgentClusterInstall not set for ClusterDeployment %s", clusterDeployment.Name)
+	// ignore unsupported platforms
+	if !isSupportedPlatform(clusterDeployment) {
```

Comment:
I noticed the first conditional in this function is `cluster.Spec.ClusterInstallRef == nil` https://github.com/openshift/assisted-service/pull/3974/files#diff-40883156aab921707c5a73dfa3b451bbd3accb4c0636d715d0a7768db8904b57R597

Do you think if we switch the order like this that the check in this function is still needed? 
Or vice-versa is the [initial check](https://github.com/openshift/assisted-service/pull/3974/files#diff-40883156aab921707c5a73dfa3b451bbd3accb4c0636d715d0a7768db8904b57R130) needed if it's checked in the function?